components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java

```diff
@@ -73,9 +81,22 @@ public static String getUsernameFromContext() {
      */
     public static URI buildURIForBody(String endpoint) {
 
-        String tenantQualifiedRelativePath =
-                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
-        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + endpoint, true, true);
+        String url = "";
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            String context = SERVER_API_PATH_COMPONENT + endpoint;
+            try {
+                url = ServiceURLBuilder.create().addPath(context).build().getRelativePublicURL();
+            } catch (URLBuilderException e) {
+                String errorDescription = "Server encountered an error while building URLs in tenant qualified mode.";
+                LOG.error(errorDescription);
```

Comment:
We don't need to log here. It will be handled by the layer which catches the APIError